### PR TITLE
Feature/add linting to build

### PR DESCRIPTION
**Pull request by alexkeating**
## Linear Ticket

https://linear.app/govrn/issue/PRO-438/start-removing-linting-warnings

## Description

Add max warnings on all apps and libs

### CODE REVIEW

File: apps/protocol-frontend/.eslintrc.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "extends": ["plugin:@nrwl/nx/react", "../../.eslintrc.json"],
-  "ignorePatterns": ["!**/*", "*.d.ts"],
+  "ignorePatterns": ["!**/*", "*.d.ts", "abnf.ts"],
```
Comment (jonathanprozzi) [inline apps/protocol-frontend/.eslintrc.json:3]: Why the ignore on the specific file? Mainly curious -- I saw a lot of warnings were coming from there

**Review by jonathanprozzi — APPROVED** [review]
Looks great -- seeing the CI issue on running `protocol-frontend-e2e:lint` but nothing specified with the error. Noticing that locally too

Changes are great though. We can merge in as soon as the CI issue is resolved!

File: apps/protocol-frontend/src/components/ContributionTypesTable.tsx [hunk 2]
```diff
@@ -59,7 +59,7 @@ const ContributionTypesTable = ({
         date_of_submission: contributionType.date_of_submission,
         date_of_engagement: contributionType.date_of_engagement,
       })),
-    [],
+    [], // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (amrro) [inline apps/protocol-frontend/src/components/ContributionTypesTable.tsx:62]: May I ask why disabling? 
Shouldn't this react to unique contributions changes?

File: apps/protocol-frontend/src/components/MyAttestationsTable.tsx [hunk 3]
```diff
@@ -137,28 +138,33 @@ const MyAttestationsTable = ({
...
-                            <IoArrowUp aria-label="sorted-ascending" />
-                          )
-                        ) : null}
-                      </chakra.span>
-                    </Th>
-                  ))}
-                </Tr>
-              ))}
+              {headerGroups.map(
+                (headerGroup: HeaderGroup<MyAttestationsTableType>) => (
```
Comment (amrro) [inline apps/protocol-frontend/src/components/MyAttestationsTable.tsx:142]: Well Done!

But I am not noticing we have like many of this: same structure, same style but different data. A separate component could handle this precedents. right?

**Review by amrro — COMMENTED** [review]
Your effort is making the project type-safer and more mature!

I suggested some minor tweaks in the same direction.

File: apps/protocol-frontend-e2e/project.json [hunk 4]
```diff
@@ -19,7 +19,8 @@
       "executor": "@nrwl/linter:eslint",
       "outputs": ["{options.outputFile}"],
       "options": {
-        "lintFilePatterns": ["apps/protocol-frontend-e2e/**/*.{js,ts}"]
+        "lintFilePatterns": ["apps/protocol-frontend-e2e/**/*.{js,ts}"],
+        "maxWarnings": 7
       }
     }
   },
```
[the remark this answers is not in the document]
Comment (amrro) [reply apps/protocol-frontend-e2e/project.json:26]: I might do some reformatting at #248

File: apps/protocol-frontend/src/components/MintModal.tsx [hunk 5]
```diff
@@ -59,14 +59,16 @@ const MintModal = ({ contributions }: MintModalProps) => {
...
 
       // Mint successfully stored contributions in IPFS.
-      await bulkMintContributions(
-        bulkStoreResult
-          .filter(promise => promise.status === 'fulfilled')
-          .map(result => ({
+      const bulkContributions = [];
+      for (const result of bulkStoreResult) {
+        if (result.status === 'fulfilled') {
+          bulkContributions.push({
```
[the remark this answers is not in the document]
Comment (alexkeating) [reply apps/protocol-frontend/src/components/MintModal.tsx]: I would disagree. I find the filter, reduce and map harder to read, harder to type and prone to less efficient outcomes than a simple for loop.

File: apps/protocol-frontend/src/components/ContributionTypesTable.tsx (hunk 2 above)
Comment (alexkeating) [reply apps/protocol-frontend/src/components/ContributionTypesTable.tsx:62]: An empty array means the useEffect is called once and it is a common practice. We won't run into the same issues as a useEffect with partial dependencies

File: apps/protocol-frontend/src/components/MyAttestationsTable.tsx (hunk 3 above)
Comment (alexkeating) [reply apps/protocol-frontend/src/components/MyAttestationsTable.tsx:142]: This pr is purely for removing linting issues and enforcing the current state of our linter warnings in the build. It isn't meant to go beyond that such as changing functionality.

File: apps/protocol-frontend-e2e/project.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexkeating) [reply apps/protocol-frontend-e2e/project.json:26]: Please do not bundle formatting changes into that pr. If anything we should do it in a separate pr and like you mentioned yesterday we should enforce it in the build.

File: apps/protocol-frontend/src/components/HomeShell.tsx [hunk 6]
```diff
@@ -64,27 +64,24 @@ const HomeShell = () => {
...
-                marginBottom={4}
-              >
-                <CreateUserForm />
-              </Flex>
-            ) : (
-              <CreateWaitlistUserForm />
-            )}
-          </>
-        )}
+        {createProfileSteps === 2 &&
```
[the remark this answers is not in the document]
Comment (alexkeating) [reply apps/protocol-frontend/src/components/HomeShell.tsx:67]: This pr is purely for removing linting issues and enforcing the current state of our linter warnings in the build. It isn't meant to go beyond that such as refactoring these values.

File: apps/protocol-frontend/.eslintrc.json (hunk 1 above)
Comment (alexkeating) [reply apps/protocol-frontend/.eslintrc.json:3]: This is code we copied from siwe. We shouldn't be modifying it and should eventually remove it.

File: apps/protocol-frontend/src/components/MyAttestationsTable.tsx (hunk 3 above)
Comment (alexkeating) [reply apps/protocol-frontend/src/components/MyAttestationsTable.tsx:142]: But, I totally aggree a lot of these headers and tables can be mode more reusable and modular. We can definitely handle in another ticket

**Comment by alexkeating** [comment]
@amrro @jonathanprozzi Thank you for the reviews!

**Pull request merged**